### DISCUSSION TITLE
Nested back relation expansion

### DISCUSSION THREAD

**Discussion by sphade** [pocketbase/pocketbase#4174] [Q&A]
Given that i have collection-A, collection-B and collection-C
collection-B has a field in it records called collection-A-id, linking it to collection-A
collection-C has a field in it records called collection-B-id, linking it to collection-B
i can use back relation expansion to get every record in collection-B related to a specific record in collection-A : 'collection-B(collection-A)
how can i do the same to get every record related to a specific record in collection-B which is also related to a specify field in collection-A i.e get a field in collection-A with all his related fields in collection-B and all collection-B related fields in collection-C

**Comment by ganigeorgiev** [comment]
I'm not sure that I understand your structure but chaining with dot-notation should work for expanding nested back relations, something like:

```
collectionB(A_rel).collectionC(B_rel)
```

If that is not what you are asking, please try to provide a more clear description of your structure with the exact name of the fields and/or export your current collections configuration as JSON (_Admin UI > Settings > Export collections_).

**Reply by sphade** [reply]
my collections configuration is very large at this point, so i don't think exporting it might help you, but what am trying to achive is this.
i have three collections called courses, chapters and muxData,
courses does not have any direct relation to chapters or muxData,
chapters has a relation to courses with the field courseId, (as a course can have many chapters)
muxData has a relation to chapters collection called chapterId  (as a chapter can have only one mux data)
i want to get a course, with all it chapters and with all it muxData

**Reply by sphade** [reply]
i have this
await pb.collection('courses').getOne("courseId", {
				expand: 'chapters(course)'
			}); 
but how do i get the muxData of the chapters along as well

**Reply by ganigeorgiev** [reply]
Then in this case something like below should work:

```js
expand: 'chapters(courseId).muxData(chapterId)'
```

Also make sure to check that your client request satisfy the View API rules of the `chapters` and `muxData` collections.

_Note that your code snippet doesn't correspond to the name of the fields in your description - course VS courseId. Make sure to replace in the above example `courseId` and `chapterId` with the name of the corresponding relation fields._
